core/src/main/scala/cats/syntax/applicative.scala

```diff
@@ -17,3 +19,8 @@ final class ApplicativeOps[F[_], A](private val fa: F[A]) extends AnyVal {
   def unlessA(cond: Boolean)(implicit F: Applicative[F]): F[Unit] = F.unlessA(cond)(fa)
   def whenA(cond: Boolean)(implicit F: Applicative[F]): F[Unit] = F.whenA(cond)(fa)
 }
+
+final class ApplicativeByNameOps[F[_], A](private val fa: () => F[A]) extends AnyVal {
```

Comment:
Perhaps a minor thing, but wouldn't it better to keep constructors of *-`Ops` classes private to the `syntax` package? I know, not all the existing classes do follow it, but perhaps it is a good idea for all new classes, wdyt?
```scala
final class ApplicativeByNameOps[F[_], A] private[syntax] (private val fa: () => F[A])
  extends AnyVal
```
I mean, it is unlikely that we want to allow users to create `Ops` files explicitly, isn't? Or is there some use case for that?